misago/legal/models.py

```diff
@@ -1 +1,113 @@
-# Empty models file that triggers migrations for this app
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.template.defaultfilters import slugify
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from misago.conf import settings
+
+
+class NoActiveLegalAgreement(ValidationError):
+    pass
+
+
+class LegalAgreementManager(models.Manager):
+    def get_current(self):
+        try:
+            return self.filter(is_active=True)
+        except self.model.DoesNotExist:
+            raise NoActiveLegalAgreement(
+                u'Please create an active LegalAgreement'
+            )
+
+
+class LegalAgreement(models.Model):
+    created_at = models.DateTimeField(default=timezone.now)
+    modified_at = models.DateTimeField(auto_now=True)
+    published_at = models.DateTimeField(blank=True, null=True)
+    key = models.SlugField(unique=True, blank=True)
+
+    TOS = 'terms_of_service'
+    PRIVACY = 'privacy_policy'
+    LEGAL_CHOICES = (
+        (TOS, 'Terms of service'),
+        (PRIVACY, 'Privacy Policy'),
+    )
+    title = models.CharField(
+        max_length=20,
+        choices=LEGAL_CHOICES,
+        default=TOS
+    )
+    text = models.TextField(
+        verbose_name=_('content'),
+        blank=True
+    )
+    is_active = models.BooleanField(
+        default=False,
+        verbose_name=_('active/publish'),
+        help_text=_(
+            u'Only one terms of service is allowed to be active'
+        )
+    )
+
+    objects = LegalAgreementManager()
+
+    def __init__(self, *args, **kwargs):
+        super(LegalAgreement, self).__init__(*args, **kwargs)
+        self.old_is_active = self.is_active
+
+    def __str__(self):
+        return self.title + "-" + self.created_at.strftime("%Y-%m-%d")
+
+    def has_user_agreed_latest_legal(self, user, legal):
+        return UserAgreementAcceptance.objects.filter(
+            legal_agreement=LegalAgreement.objects.get_current().get(title=legal),
+            user=user,
+        ).exists()
+
+    def accept(self, user):
```

Comment:
Misago is not doing fat models. Please extract business logic from models to utility modules. I know this goes against the piece of wisdom passed around to have as much business logic in fat models, but this approach doesn't scale for complex projects that are implementing a lot of logic around relatively few models.